### PR TITLE
Technical documentation about integration with CardinalCommerce

### PR DESCRIPTION
**Pull request by viktym**
## Purpose of this pull request
This pull request (PR) describes how basic integration with CardinalCommerce can be leveraged to implement 3D Secure 2.0 for different payment integrations, supported by [CardinalCommerce](https://www.cardinalcommerce.com/)
- Internal Story - [MAGETWO-99736](https://jira.corp.magento.com/browse/MAGETWO-99736)

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.3/payments-integrations/

## Links to Magento source code

- https://github.com/magento/magento2/tree/2.3-develop/app/code/Magento/CardinalCommerce
- https://github.com/magento/magento2/tree/2.3-develop/app/code/Magento/AuthorizenetCardinal

## Requested Reviewers

@joni-jones - Payment architect 
Mikhail Chernenko - Product Owner

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/devdocs?pullRequest=5011) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/devdocs?pullRequest=5011) before we can accept your contribution.<br/>

**Comment by magento-engcom-team** [comment]
@viktym thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by viktym** [comment]
> Looks good. There is a spelling error in the image. Bottom right, "authentication".
> And if you can make sure the image is optimized for file size, that would be great.

@dobooth  I fixed this spelling error and made image optimizing

File: guides/v2.3/payments-integrations/cardinal/cardinal.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
+---
```
Comment (keharper) [inline guides/v2.3/payments-integrations/cardinal/cardinal.md]: I can't tell whether the 2.3 version is identical to the 2.2 version without downloading the files and running a diff, but spot checks look the same, except the 2.3 version doesn't have a blank line at the end. If they are the same, create a symlink between 2.2 and 2.3. 

I'm assuming the versions are the same, or are similar enough that the edits I made in 2.2 also apply to 2.3.

File: guides/v2.2/payments-integrations/cardinal/cardinal.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+The following diagram shows a simplified 3-D Secure verification flow using [Cardinal Cruise Standard](https://cardinaldocs.atlassian.net/wiki/spaces/CC/pages/7929857/Cardinal+Cruise+Standard) integration approach provided by [CardinalCommerce](https://www.cardinalcommerce.com/):
+
+![CardinalCommerce Interaction]({{ site.baseurl }}/common/images/payments-integrations/cardinal_flow.png)
+
+## Magento_CardinalCommerce module overview
+
+The *Magento_CardinalCommerce* [module](https://glossary.magento.com/module) allows you to:
+
+- start [Cardinal Consumer Authentication](https://cardinaldocs.atlassian.net/wiki/spaces/CC/pages/196642/Consumer+Authentication#ConsumerAuthentication-CardinalConsumerAuthentication) for enabling card network programs including Verified by Visa®, MasterCard SecureCode® and Identity Check®, American Express SafeKey®, Discover ProtectBuy® and Diners International® and JCB J-Secure®.
```
[the remark this answers is not in the document]
Comment (viktym) [reply guides/v2.2/payments-integrations/cardinal/cardinal.md]: Regarding tradmark, I'm not sure. I took info from this source https://cardinaldocs.atlassian.net/wiki/spaces/CC/pages/196642/Consumer+Authentication?showComments=true&showCommentArea=true

File: guides/v2.2/payments-integrations/cardinal/cardinal.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+- handle specific return values for [Cardinal Consumer Authentication](https://cardinaldocs.atlassian.net/wiki/spaces/CC/pages/196642/Consumer+Authentication#ConsumerAuthentication-CardinalConsumerAuthentication) on backend and storefront
+
+## Payment method module integration with Magento_CardinalCommerce
+
+List of compatible with [CardinalCommerce](https://www.cardinalcommerce.com/) payment gateways you can find [here](https://www.cardinalcommerce.com/partners/gateways)
+
+### CardinalCommerce configuration for payment method
+
+You need to add configuration parameter that will enable [Cardinal Consumer Authentication](https://cardinaldocs.atlassian.net/wiki/spaces/CC/pages/196642/Consumer+Authentication#ConsumerAuthentication-CardinalConsumerAuthentication) in the `config.xml` and `system.xml` files  of your [payment method](https://glossary.magento.com/payment-method) module:
```
[the remark this answers is not in the document]
Comment (viktym) [reply guides/v2.2/payments-integrations/cardinal/cardinal.md]: Actually no. Magento provides only one payment method integration with CardinalCommerce. It's integration with Authorize.net. This document describes how to create integrations CardinalCommerce with other payment methods.

File: guides/v2.3/payments-integrations/cardinal/cardinal.md (hunk 1 above)
Comment (viktym) [reply guides/v2.3/payments-integrations/cardinal/cardinal.md]: Actually, it's a symlink

File: guides/v2.2/payments-integrations/cardinal/cardinal.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+
+The [Cardinal Cruise Standard](https://cardinaldocs.atlassian.net/wiki/spaces/CC/pages/7929857/Cardinal+Cruise+Standard) integration is purely a JavaScript approach that is all encompassing. When enabling this approach for [Cardinal Consumer Authentication](https://cardinaldocs.atlassian.net/wiki/spaces/CC/pages/196642/Consumer+Authentication#ConsumerAuthentication-CardinalConsumerAuthentication), this integration will handle the device data collection, initiating the transaction for [CCA](https://cardinaldocs.atlassian.net/wiki/spaces/CC/pages/196642/Consumer+Authentication#ConsumerAuthentication-CardinalConsumerAuthentication), presenting the authentication session if required, and returning the results of authentication once completed. This is a simple and easy approach is recommended integration approach for [CCA](https://cardinaldocs.atlassian.net/wiki/spaces/CC/pages/196642/Consumer+Authentication#ConsumerAuthentication-CardinalConsumerAuthentication).
```
[the remark this answers is not in the document]
Comment (viktym) [reply guides/v2.2/payments-integrations/cardinal/cardinal.md]: Done

File: guides/v2.2/payments-integrations/cardinal/cardinal.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (viktym) [reply guides/v2.2/payments-integrations/cardinal/cardinal.md]: Bulleted items updated with capital letters

File: guides/v2.2/payments-integrations/cardinal/cardinal.md [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+## Magento_CardinalCommerce module overview
+
+The *Magento_CardinalCommerce* [module](https://glossary.magento.com/module) allows you to:
+
+- start [Cardinal Consumer Authentication](https://cardinaldocs.atlassian.net/wiki/spaces/CC/pages/196642/Consumer+Authentication#ConsumerAuthentication-CardinalConsumerAuthentication) for enabling card network programs including Verified by Visa®, MasterCard SecureCode® and Identity Check®, American Express SafeKey®, Discover ProtectBuy® and Diners International® and JCB J-Secure®.
+- handle specific return values for [Cardinal Consumer Authentication](https://cardinaldocs.atlassian.net/wiki/spaces/CC/pages/196642/Consumer+Authentication#ConsumerAuthentication-CardinalConsumerAuthentication) on backend and storefront
+
+## Payment method module integration with Magento_CardinalCommerce
+
+List of compatible with [CardinalCommerce](https://www.cardinalcommerce.com/) payment gateways you can find [here](https://www.cardinalcommerce.com/partners/gateways)
```
[the remark this answers is not in the document]
Comment (viktym) [reply guides/v2.2/payments-integrations/cardinal/cardinal.md]: Done

File: guides/v2.2/payments-integrations/cardinal/cardinal.md [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+        });
+    };
+});
+```
+
+Once the response JWT is received after consumer authentication, you will need to send it to your backend to verify and extract the results. In the example above response JWT is added to payment additional data and passed to backend along with them.
+
+### CCA Results Extracting And Validation On Backend
+
+Cardinal Consumer Authentication results can be extracted from CardinalCommerce response JWT with `\Magento\CardinalCommerce\Model\Response\JwtParserInterface`. Basic implementation of this interface includes response JWT signature validation, and validation such params as `ActionCode`, `ErrorNumber`, `ECIFlag`. Detailed information about these params you can find in [API Reference](https://cardinaldocs.atlassian.net/wiki/spaces/CC/pages/98315/Response+Objects)
```
[the remark this answers is not in the document]
Comment (viktym) [reply guides/v2.2/payments-integrations/cardinal/cardinal.md]: Done

File: guides/v2.2/payments-integrations/cardinal/cardinal.md [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+                ]
+            ];
+        }
+
+        return $data;
+    }
+}
+```
+
+Depending from requirements of your payment gateway some of response fields you should include in the transaction request or perform separate request based on these values.
```
[the remark this answers is not in the document]
Comment (viktym) [reply guides/v2.2/payments-integrations/cardinal/cardinal.md]: Done

File: guides/v2.2/payments-integrations/cardinal/cardinal.md [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+
+In our example `ECIFlag` and `CAVV` values were included in the transaction request to Authorize.Net
```
[the remark this answers is not in the document]
Comment (viktym) [reply guides/v2.2/payments-integrations/cardinal/cardinal.md]: Done

File: guides/v2.2/payments-integrations/cardinal/cardinal.md [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+
+Then you can expect to see an additional field with cardholder authentication verification response code in the response from your payment gateway. It's letting you know whether the information got back to the issuer okay. If the issuer recognizes this data as matching whatever they recorded earlier in the transaction when the cardholder was authenticating, they’ll respond with a successful code in this field.
```
[the remark this answers is not in the document]
Comment (viktym) [reply guides/v2.2/payments-integrations/cardinal/cardinal.md]: Done

File: guides/v2.3/payments-integrations/cardinal/cardinal.md (hunk 1 above)
Comment (viktym) [reply guides/v2.3/payments-integrations/cardinal/cardinal.md]: Symlink created

**Comment by viktym** [comment]
@keharper I've made corrections, please review. Thanks for help

**Comment by viktym** [comment]
@keharper @dobooth should I need to perform any other corrections?

**Comment by dobooth** [comment]
running tests

**Comment by meker12** [comment]
@keharper  Missing `whatsnew` text required for new topics.

**Comment by keharper** [comment]
@dobooth worked on this PR

**Pull request merged**